packages/react-reconciler/src/ReactFiberLane.js

```diff
@@ -365,6 +365,37 @@ export function getNextLanes(root: FiberRoot, wipLanes: Lanes): Lanes {
     }
   }
 
+  // Check for entangled lanes and add them to the batch.
+  //
+  // A lane is said to be entangled with another when it's not allowed to render
+  // in a batch that does not also include the other lane. Typically we do this
+  // when multiple updates have the same source, and we only want to respond to
+  // the most recent event from that source.
+  //
+  // Note that we apply entanglements *after* checking for partial work above.
+  // This means that if a lane is entangled during an interleaved event while
+  // it's already rendering, we won't interrupt it. This is intentional, since
+  // entanglement is usually "best effort": we'll try our best to render the
+  // lanes in the same batch, but it's not worth throwing out partially
+  // completed work in order to do it.
```

Comment:
This comment seems at odds with how `useMutableSource` relies no entanglements to work.

Reply:
For `useMutableSource` it's ok because the `markRootEntangled` call happens in the commit phase, where there's no partial work. I wrote this comment here as a reminder for the future.

The other case we have planned for entanglement is skipping intermediate `useTransition` updates, where you don't want to interrupt.